experiments/signals.py

```diff
@@ -17,3 +17,16 @@ def on_datapoint_creation(sender, instance, *args, **kwargs):
             instance.number = last_datapoint.number + 1
         else:
             instance.number = 1
+
+
+@receiver(pre_save, sender=ParticipantSession)
+def on_participant_session_creation(sender, instance, *args, **kwargs):
+    """
+    Add an incremental session_id number when saving a session starting from 1
+    """
+    if not instance.subject_id:
+        experiment = instance.experiment
+        experiment_sessions = experiment.session_set.all()
+        new_id = len(experiment_sessions) + 1
```

Comment:
It's probably better to do ``experiment.session_set.count()``, as that will do a more lightweigt SQL query.

Reply:
This should be fixed by now. Also session_set was incorrect as django creates it as Experiment.participantsession_set